### PR TITLE
Migrate define ITs to BDD

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

Where possible, we would like our Graql `define` tests to be migrated to BDD.

This PR forms part of the work towards #5719 .

## What are the changes implemented in this PR?

This PR will gradually delete test cases from GraqlDefineIT as they are tested in BDD.

This PR should NOT be merged until verification is updated with the new tests, which is the desired outcome of https://github.com/graknlabs/verification/pull/53

### CODE REVIEW

File: test/integration/graql/query/GraqlDefineIT.java [hunk 1]
```diff
@@ -296,109 +121,20 @@ public void whenDefiningARuleUsingCoreAPI_ruleIsPersistedCorrectly(){
...
 
-    @Test
-    public void whenATypeHasNonExistentResource_weThrow() {
-        exception.expect(GraqlSemanticException.class);
-        exception.expectMessage("nothing");
-        tx.execute(Graql.define(type("blah this").sub("entity").has("nothing")));
-    }
-
     @Test
     public void whenDefiningMetaType_Throw() {
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDefineIT.java:132]: this is migratable - this is trying to define a subtype of "thing" directly, which is not allowed :)

File: test/integration/graql/query/GraqlDefineIT.java [hunk 2]
```diff
@@ -407,104 +143,14 @@ public void whenDefiningAThing_Throw() {
...
-        exception.expectMessage(anyOf(
-                is(GraqlSemanticException.defineUnsupportedProperty(Graql.Token.Property.HAS.toString()).getMessage()),
-                is(GraqlSemanticException.defineUnsupportedProperty(Graql.Token.Property.VALUE.toString()).getMessage())
-        ));
-
-        tx.execute(Graql.define(var().id(id.getValue()).has("title", "Bob")));
-    }
-
-    @Test @Ignore
-    public void whenSpecifyingLabelOfAnExistingConcept_LabelIsChanged() {
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDefineIT.java:424]: i think we should open a new issue in the `graql` repository to bring this feature back: ability to change the label of a type using graql itself. You can document that the old way to do this was by awkwardly combining an ID with type(), which look like two setters or two getters and not a mixture. We need to rethink how this should be properly designed again in the future, for now it is banned via graql and only allowed via concept API

File: test/integration/graql/query/GraqlDefineIT.java [hunk 3]
```diff
@@ -407,104 +143,14 @@ public void whenDefiningAThing_Throw() {
...
-        exception.expect(GraknException.class);
-
-        tx.execute(Graql.define(
-                type("marriage").sub(Graql.Token.Type.RELATION).relates("husband").relates("wife"),
-                type("wife").sub(Graql.Token.Type.ENTITY)
-        ));
-    }
-
-    private boolean schemaObjectsExist(Statement... vars){
+    /*private boolean schemaObjectsExist(Statement... vars){
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDefineIT.java:153]: eh?

File: test/integration/graql/query/GraqlDefineIT.java [hunk 4]
```diff
@@ -407,104 +143,14 @@ public void whenDefiningAThing_Throw() {
...
-        Pattern when = Graql.parsePattern("$x isa entity;");
-        Pattern then = Graql.parsePattern("$x isa entity;");
-        Statement vars = type("my-rule").when(when).then(then);
-        tx.execute(Graql.define(vars));
-
-        assertNotNull(tx.getRule("my-rule"));
-    }
-
-    @Test
-    public void whenDefiningARelation_SubRoleDeclarationsCanBeSkipped_EvenWhenRoleInReferredToInOtherContexts() {
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDefineIT.java:482]: we got a test like this?

File: test/integration/graql/query/GraqlDefineIT.java (hunk 3 above)
Comment (flyingsilverfin) [reply test/integration/graql/query/GraqlDefineIT.java:153]: clean up comments

File: test/integration/graql/query/GraqlDefineIT.java (hunk 1 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDefineIT.java:132]: Ok, done!

The new test is: `Scenario: define new meta-type ('sub thing') throws`

File: test/integration/graql/query/GraqlDefineIT.java (hunk 2 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDefineIT.java:424]: Sounds good

https://github.com/graknlabs/graql/issues/131

File: test/integration/graql/query/GraqlDefineIT.java (hunk 4 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDefineIT.java:482]: It tests that we don't need `sub role` to define a role. Given that we're phasing out `sub role`, the test has become redundant.

File: .circleci/config.yml [hunk 5]
```diff
@@ -87,6 +89,26 @@ jobs:
...
+          command: bazel test //server/... --test_output=errors
+
+  test-graql:
+    machine: true
+    working_directory: ~/grakn
+    steps:
+      - install-bazel-linux-rbe
+      - checkout
+      - run-bazel-rbe:
+          command: bazel test //graql/... --test_output=errors
```
Comment (haikalpribadi) [inline .circleci/config.yml:108]: I thought @flyingsilverfin had added this in a previous PR? Why does it need to be added again?

File: .circleci/config.yml [hunk 6]
```diff
@@ -508,6 +523,14 @@ workflows:
...
             branches:
               ignore: grakn-release-branch
+      - test-server:
+          filters:
+            branches:
+              ignore: grakn-release-branch
+      - test-graql:
+          filters:
+            branches:
+              ignore: grakn-release-branch
```
Comment (haikalpribadi) [inline .circleci/config.yml:533]: And this too ...

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -1,10 +1,9 @@
...
 
-/dependencies/graknlabs/  @haikalpribadi
-/dependencies/maven/      @haikalpribadi @marco-scoppetta @lolski
+/.circleci/             @haikalpribadi @lolski
+/.circleci/windows/     @vmax
+/.github/               @haikalpribadi
+/config/                @lolski @vmax
+/daemon/                @haikalpribadi @lolski
+/dependencies/          @haikalpribadi @flyingsilverfin
+/graql/reasoner/        @flyingsilverfin
```
Comment (haikalpribadi) [inline .github/CODEOWNERS:9]: again, wasn't this in @flyingsilverfin previous PR?

Comment (alexjpwalker) [reply .github/CODEOWNERS:9]: We talked this over with Ganesh and established that it was a Github bug.

I'm closing this PR and creating a clone of it at https://github.com/graknlabs/grakn/pull/5756 .

**Pull request closed without merging**